File: code/community/1306404111066861570/handling-regional-pronunciation-deepgram-tts.py
import os
import requests

# Set your Deepgram API key as an environment variable before running the script
API_KEY = os.getenv('DEEPGRAM_API_KEY')

# Define the URL for the TTS service
url = 'https://api.deepgram.com/v1/speak'

# Function to convert text to speech
# Modify names creatively to alter pronunciation
text_payload = {
    "text": "Hello, this is a test pronunciation of the name SpellingAlteration."
}

headers = {
    'Authorization': f'Token {API_KEY}',
    'Content-Type': 'application/json'
}

response = requests.post(url, json=text_payload, headers=headers)

if response.status_code == 200:
    print('Request successful!')
    print('Audio response URL:', response.json().get('url'))
else:
    print('Request failed with status code:', response.status_code)

File: code/community/1306404111066861570/handling-regional-pronunciation-deepgram-tts.js
const axios = require('axios');

// Make sure to set your API key as an environment variable
const apiKey = process.env.DEEPGRAM_API_KEY;

const url = 'https://api.deepgram.com/v1/speak';

const textPayload = {
  text: 'Hello, this is a test pronunciation of the name SpellingAlteration.'
};

axios.post(url, textPayload, {
  headers: {
    'Authorization': `Token ${apiKey}`,
    'Content-Type': 'application/json'
  }
})
.then(response => {
  console.log('Request successful!');
  console.log('Audio response URL:', response.data.url);
})
.catch(error => {
  console.error('Request failed:', error);
});